src/hooks/useAlerts.ts

```diff
@@ -1,47 +1,43 @@
 /**
- * This custom React Hook provides functions to manage alerts in your application.
- * It allows you to add and remove alerts with specific titles, variants, and unique keys.
+ * This Hook provides functions to manage alerts.
+ * It allows you to add and remove alerts using PatternFly's AlertProps and unique IDs.
  * @module useAlerts
  */
 import React from 'react';
-import { AlertProps as PfAlertProps } from '@patternfly/react-core';
-
-interface AlertProps extends PfAlertProps {
-  key: React.Key;
-}
+import { AlertProps } from '@patternfly/react-core';
 
 const useAlerts = () => {
-  const [alert, setAlerts] = React.useState<AlertProps[]>([]);
+  const [alerts, setAlerts] = React.useState<AlertProps[]>([]);
+
+  let alertCounter = 1;
 
   /**
    * Add an Alert
    *
-   * This function adds an alert to the list of alerts.
+   * This function adds an alert to the list of alerts, generating a unique ID.
    *
-   * @param {AlertProps} options - An object containing the properties of the alert.
-   * @param {string} options.title - The title or content of the alert.
-   * @param {AlertProps['variant']} options.variant - The variant or style of the alert (e.g., 'success', 'danger').
-   * @param {React.Key} options.key - A unique key to identify the alert.
+   * @param {AlertProps} options - The properties of the alert.
    */
   const addAlert = (options: AlertProps) => {
-    setAlerts(prevAlerts => [...prevAlerts, options]);
+    const uniqueId = `alert-${alertCounter++}`;
```

Comment:
@cdcabrera I used an alertCounter here to generate the ID mainly for its simplicity, but also due to the risk of id collision of getUniqueId from PF. I mean, this use case is really simple and maybe I shouldn't worry about this level of complexity here. What's your opinion on the matter ?

Reply:
So this is where default properties would normally come in at the component level.
> Generally, I'm not a fan of sequential identifiers, history has shown me there are too many things that end up going weird.

We're attempting to bypass the need to write a centralized alert component by modifying state only. The preference is we do those defaults at the component level. What you have would sort-of work, but it leans towards modifying state. It's not wrong, we just need to be consistent since right now the code is blending state modifications and component defaults.
> This state modification behavior is why v1 code wraps many PF components, PF components don't always come with state, some do, but not all.

So, we need to pick a path, picking a path includes.... (I can go either path on this, so no wrong choice, but you have to make that decision)

**"fix what we can and get the tests working" aka continue what you started**
- we continuing your work to separate out the `useAlerts` hook (we probably will end up doing this for each hook in case they cover similar future modifications)
- we continue minimally modifying `useAlerts` which includes
   - continuing to move over to removing "key" and flipping to "id"
   - but we modify where the `getUniqueId` is placed, we move it to the component level instead, in your views
      ```
      <AlertGroup isToast isLiveRegion>
        {alerts.map(({ id, variant, title }) => (
          <Alert
            timeout={8000}
            onTimeout={() => id && removeAlert(id)}
            variant={AlertVariant[variant || 'info']}
            title={title}
            actionClose={
              <AlertActionCloseButton
                title={title as string}
                variantLabel={`${variant} alert`}
                onClose={() => id && removeAlert(id)}
              />
            }
            key={id || getUniqueId()}
          />
        ))}
      </AlertGroup>
      ```

**picking path "fix all alerts now" includes**

 - we create a centralized "alert component" similar to what we created in v1 and co-locate a new component possibly with the hook it needs to function
    - https://github.com/quipucords/quipucords-ui/blob/main/src/components/toastNotificationsList/toastNotificationsList.js
- we update all of our views with this new component and remove the unique activations of alerts per view
- we update our tests